### PR TITLE
Version support in github actions vs pyproject.toml

### PR DESCRIPTION
**Pull request by sierra-moxon**

### CODE REVIEW

File: linkml_runtime/__init__.py [hunk 1]
```diff
@@ -4,6 +4,11 @@
 import rdflib_shim
 shim = rdflib_shim.RDFLIB_SHIM
 
+try:
```
Comment (cmungall) [inline linkml_runtime/__init__.py:7]: can you add a comment to say why this is necessary?

File: tests/test_utils/test_version.py [hunk 2]
```diff
@@ -0,0 +1,3 @@
+import linkml_runtime
+
+print(linkml_runtime.__version__)
```
Comment (cmungall) [inline tests/test_utils/test_version.py:3]: can you make it an actual test (and avoid prints)?

**Review by cmungall — APPROVED** [review]
let's try it!

File: tests/test_utils/test_version.py (hunk 2 above)
Comment (sujaypatil96) [reply tests/test_utils/test_version.py]: Updated the unit test.

File: linkml_runtime/__init__.py (hunk 1 above)
Comment (sujaypatil96) [reply linkml_runtime/__init__.py:10]: Added a comment clarifying why we need `importlib-metadata`.

**Review by sujaypatil96 — APPROVED** [review]
@sierra-moxon: So it turns out `importlib-metadata` isn't bundled in native Python for versions `<3.8`, so I've added it as a dev dependency in the pyproject.toml file if the version being tested is `<3.8`.

@cmungall: Perhaps we can create a release from the main branch to see how this solution works. We created releases from this branch already and it works like a charm.

**Comment by cmungall** [comment]
go for it!

**Pull request merged**